src/main/webapp/app/entities/competency.model.ts

```diff
@@ -83,6 +83,34 @@ export class Competency extends CourseCompetency {
     }
 }
 
+export class CompetencyLearningObjectLink {
+    competency?: CourseCompetency;
```

Comment:
can there be a CompetencylearningObjectLink without a competency?

Reply:
Yes, since we sadly do not use DTOs for the exercise and competency endpoints yet, a competency might have its links stored but these links do not contain the competency again to avoid circular dependencies.